Public/Src/Engine/Scheduler/FileSystem/FileSystemView.cs

```diff
@@ -151,7 +151,9 @@ public Possible<PathExistence> TryEnumerateDirectory(AbsolutePath path, FileSyst
             PathExistence existence;
             if (PathExistenceCache.TryGetValue(path, out entry) && entry.TryGetExistence(mode, out existence))
             {
-                if (existence != PathExistence.ExistsAsDirectory)
+                if (existence == PathExistence.Nonexistent
+                    || (existence == PathExistence.ExistsAsFile 
+                        && !FileUtilities.IsDirectorySymlinkOrJunction(path.ToString(PathTable))))
                 {

```

Comment:
>{ [](start = 16, length = 1)

Update flags IsDirectorySymlink and CheckedIsDirectorySymlink to enable caching of this check.

PathExistenceCache.AddOrUpdate(path, false,
                            (key, data) => { throw Contract.AssertFailure("Entry should already be added for path"); },
                            (key, data, oldValue) => oldValue.SetFlag(FileSystemEntryFlags.CheckedIsDirectorySymlink | (isDirectorySymlink ?  FileSystemEntryFlags.IsDirectorySymlink : FileSystemEntryFlags.None)));